src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/source_filters_table.js

```diff
@@ -1,128 +1,192 @@
-                  {
-                    markup: controlsHtml,
-                    scope: rowScope
-                  }
-                ]);
-              });
-              // Update the tab count
-              find($scope.$parent.editSections, { index: 'sourceFilters' }).count = $scope.rows.length;
-            }
-          });
-        }
+  constructor(props) {
+    super(props);
 
-        all() {
-          return this.$scope.indexPattern.sourceFilters || [];
-        }
+    this.state = {
+      filterToDelete: undefined,
+      isDeleteConfirmationModalVisible: false,
+      isSaving: false,
+      filters: [],
+      ...DEFAULT_TABLE_OF_RECORDS_STATE,
+    };
+  }
 
-        delete(filter) {
-          const doDelete = () => {
-            if (this.editing === filter) {
-              this.editing = null;
-            }
-
-            this.$scope.indexPattern.sourceFilters = without(this.all(), filter);
-            return this.save();
-          };
-
-          const confirmModalOptions = {
-            confirmButtonText: 'Delete',
-            onConfirm: doDelete,
-            title: 'Delete source filter?'
-          };
-          confirmModal('', confirmModalOptions);
-        }
+  componentWillMount() {
+    this.putFiltersInState();
+  }
 
-        create() {
-          const value = this.newValue;
-          this.newValue = null;
-          this.$scope.indexPattern.sourceFilters = [...this.all(), { value }];
-          return this.save();
-        }
+  putFiltersInState = () => {
+    const filters = this.props.indexPattern.sourceFilters;
 
-        save() {
-          this.saving = true;
-          this.$scope.indexPattern.save()
-            .then(() => this.editing = null)
-            .catch(notify.error)
-            .finally(() => this.saving = false);
-        }
+    this.setState({
+      filters,
+      ...this.computeTableState(this.state.criteria, this.props, filters)
+    });
+  }
+
+  onDataCriteriaChange = criteria => {
+    this.setState(this.computeTableState(criteria));
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.filterFilter !== nextProps.filterFilter) {
+      this.setState(this.computeTableState(this.state.criteria, nextProps));
+    }
+  }
+
+  computeTableState(criteria, props = this.props, filters = this.state.filters) {
+    let items = filters;
+    if (props.filterFilter) {
+      const filterFilter = props.filterFilter.toLowerCase();
+      items = items.filter(filter => filter.value.toLowerCase().includes(filterFilter));
+    }
+
+    return getTableOfRecordsState(items, criteria);
+  }
+
+  startDeleteFilter = filter => {
+    this.setState({ filterToDelete: filter, isDeleteConfirmationModalVisible: true });
+  }
+
+  hideDeleteConfirmationModal = () => {
+    this.setState({ filterToDelete: undefined, isDeleteConfirmationModalVisible: false });
+  }
+
+  deleteFilter = async () =>  {
+    const { indexPattern, onAddOrRemoveFilter } = this.props;
```

Comment:
This deletes filters with the same `value`:

![feb-16-2018 11-38-29](https://user-images.githubusercontent.com/1965714/36325865-f62e24c0-130d-11e8-8fd3-d0c2495b65d0.gif)